include/ros2_logging_fmt/ros2_logging_fmt.hpp

```diff
@@ -15,49 +16,119 @@ class Logger
   Logger(rclcpp::Logger rclcpp_logger): rclcpp_logger_(rclcpp_logger)
   {}
```

Comment:
What happens if someone initializes like this and then calls the function that does not provide a clock? The clock shared_ptr will be nullptr and probably throw an exception. Would it be better to make it a `std::optional<std::shared_ptr<Clock>>` and then when you access the value of the optional it'll throw instead of segfault?

Reply:
I did some more research into `std::optional`.  I am currently targeting Foxy which has a [target of C++14](https://www.ros.org/reps/rep-2000.html#foxy-fitzroy-may-2020-may-2023). 

The next LTS has target of C++17 which is good but not current yet. So there is that blocker at the moment.

The other thing is not for sure what to do if the optional doesn't have a clue. Should we just print a message? We can throw an exception but not everyone uses exceptions in C++. I don't want to impose a particular design.

I am wondering if we offer too many way to initialize the logger and need to slim it down a bit. Almost 100% of my use case I'll be passing in a ROS node. What are your thoughts?